merkle/log_proofs.go

```diff
@@ -384,12 +384,15 @@ func lastNodePresent(level, ts int64) bool {
 	return b&mask != 0
 }
 
-// skipMissingLevels moves down the tree a level towards the leaves until the node exists. This
-// must terminate successfully as we will eventually reach the leaves, which are always written
-// and are at level 0. Missing nodes are intermediate nodes with one child, hence their value
-// is the same as the node lower down the tree as there is nothing to hash it with.
-func skipMissingLevels(snapshot, lastNode int64, level int, node int64) (int, int64) {
+// siblingIDSkipLevels creates a new NodeID for the supplied node, accounting for levels skipped
```

Comment:
Maybe consider flipping the info around here; the primary job is to return sibling id , an implementation detail/tweaks is that it takes into account the floating node approach of the tree. E.g:

Func returns the id of the sibling to the passed in node, taking into account the skipping of levels in a non perfect tree.

Reply:
Done.